### PR TITLE
feat: Shows Youtube and Vimeo links as embeds [cw-1393]

### PR DESCRIPTION
**Pull request by nithindavid**
## Description
This change will render the youbtube, vimeo and .mp4 urls as embeds in the article page for help center. 

**Loom**
https://www.loom.com/share/5b967b527dfb453a8e5ee15501f4c4ef?sid=71d92ebe-03e7-4c1e-a77e-c0f869f05f92

**Fixes**
https://linear.app/chatwoot/issue/CW-1393/help-center-support-video-upload-in-articles

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: spec/lib/custom_markdown_renderer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    context 'when link is a normal URL' do
+      let(:normal_url) { 'https://example.com' }
+
+      it 'renders a normal link' do
+        output = render_markdown_link(normal_url)
+        expect(output).to include('<a href="https://example.com">')
+      rescue NoMethodError => e
+        print e.message
+        # Ignore error for this test.
+      end
```
Comment (sojan-official) [inline spec/lib/custom_markdown_renderer_spec.rb:108]: this shouldn't be here

File: lib/custom_markdown_renderer.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    mp4_match = link_url.match(MP4_REGEX)
+    if mp4_match
+      embed_code = make_video_embed(link_url)
+      out(embed_code)
+      return
+    end
+
+    # If it's not YouTube or Vimeo link, render normally
+    super
+  end
```
Comment (sojan-official) [inline lib/custom_markdown_renderer.rb:47]: rather than disabling the rubocop here, we can break the method into sub methods

**Review by sojan-official — COMMENTED** [review]
minor comments

File: lib/custom_markdown_renderer.rb (hunk 2 above)
Comment (nithindavid) [reply lib/custom_markdown_renderer.rb]: @tejaswinichile the return in those cases is needed; I have pushed the refactored changes. Please check cc: @sojan-official

File: spec/lib/custom_markdown_renderer_spec.rb (hunk 1 above)
Comment (nithindavid) [reply spec/lib/custom_markdown_renderer_spec.rb]: Made the change.

**Pull request merged**